### DISCUSSION TITLE
Debugging with Pluto

### DISCUSSION THREAD

**Discussion by sophia-wright-blue** [fonsp/Pluto.jl#2627] [Questions about using Pluto] [answered]
hello - is it possible to debug with Pluto by setting breakpoints and stepping into the code 

this is possible with VSCode, but breaks for some code I'm running due to a bug in Interpreter.jl : https://github.com/JuliaDebug/JuliaInterpreter.jl/issues/574 

thank you.

**Comment by pankgeorg** [comment, marked as answer]
No, this is not possible at the moment. You _could_ theoretically disable all cells and run them one by one, but that wouldn't work very well with reactivity.

Some special UI to do this step-by-step would actually be nice!

**Comment by disberd** [comment]
@sophia-wright-blue You might want to look at the very recently released https://github.com/disberd/PlutoVSCodeDebugger.jl

For code defined inside notebooks you still have to set breakpoints manually (as written in the documentation) but I have some ideas on how to try making VSCode breakpoint on the notebook file work. Hopefully this will be a feature in a next release

**Comment by sophia-wright-blue** [comment]
this is amazing! thank you so much for letting me know @disberd and for releasing this package and for the cool video demos!

I'll keep track of your repo

in case it's helpful to you, there are a couple of issues that are WIP for debugging with VSCode:

https://github.com/JuliaDebug/JuliaInterpreter.jl/issues/574

https://github.com/julia-vscode/julia-vscode/issues/3349
